conn.go

```diff
@@ -450,13 +455,14 @@ func (c *Conn) Send(destination, contentType string, body []byte, opts ...func(*
 			C:     make(chan *frame.Frame),
 		}
 
-		err := sendDataToWriteChWithTimeout(c.writeCh, request, c.msgSendTimeout)
-		if err != nil {
-			return err
+		sendErr := sendDataToWriteChWithTimeout(c.writeCh, request, c.msgSendTimeout)
+		if sendErr != nil {
+			return sendErr
```

Comment:
(minor) It may be slightly more idiomatic to re-use the `err` variable if we don’t need to keep the value around:
```suggestion
		err := sendDataToWriteChWithTimeout(c.writeCh, request, c.msgSendTimeout)
		if err != nil {
			return err
```

Reply:
Happy with this, particularly if this is the more established pattern.